src/NuGet.Core/NuGet.Commands/RestoreCommand/Utility/MSBuildRestoreUtility.cs

```diff
@@ -515,6 +518,28 @@ private static IEnumerable<IMSBuildItem> GetItemByType(IEnumerable<IMSBuildItem>
             return items.Where(e => type.Equals(e.GetProperty("Type"), StringComparison.OrdinalIgnoreCase));
         }
 
+        /// <summary>
+        /// Return the parsed version or 1.0.0 if the property does not exist.
+        /// </summary>
+        private static NuGetVersion GetVersion(IMSBuildItem item)
+        {
+            var versionString = item.GetProperty("Version");
+            NuGetVersion version = null;
+
+            if (string.IsNullOrEmpty(versionString))
+            {
+                // Default to 1.0.0 if the property does not exist
+                version = new NuGetVersion(1, 0, 0);
+            }
+            else
+            {
+                // Snapshot versions are not allowed in .NETCore
+                version = NuGetVersion.Parse(versionString);
```

Comment:
Yes, we should fail on bad versions. The reason empty is allowed right now is because CPS needs to start passing this also, once it is fully hooked up this could be more strict.

Ideally the version would also be parsed when the project file is read so that the file/line number was added to the exception.